ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -0,0 +1,962 @@
+    command.add(TABLE_NAME, tblName);
+    command.add(COLUMN_NAME, colName);
+    command.add(PARTITION_NAME, ptName);
+    List<String> temp = new ArrayList<>();
+    addBinaryStats(columnStatisticsData, temp);
+    addLongStats(columnStatisticsData, temp);
+    addBooleanStats(columnStatisticsData, temp);
+    addStringStats(columnStatisticsData, temp);
+    addDateStats(columnStatisticsData, temp);
+    addDoubleStats(columnStatisticsData, temp);
+    addDecimalStats(columnStatisticsData, temp);
+    command.add(TBLPROPERTIES, Joiner.on(",").join(temp));
+    return command.render();
+  }
+
+  /**
+   * Parses the ColumnStatistics for all the columns in a given partition and adds the alter table update
+   * statistics command for each column.
+   *
+   * @param columnStatisticsObjList
+   * @param colName
+   * @param tblName
+   * @param ptName
+   * @param dbName
+   */
+  public List<String> getAlterTableStmtPartitionStatsColsAll(List<ColumnStatisticsObj> columnStatisticsObjList,
+      List<String> colName,
+      String tblName,
+      String ptName,
+      String dbName) {
+    List<String> alterTableStmt = new ArrayList<String>();
+    ColumnStatisticsObj[] columnStatisticsObj = columnStatisticsObjList.toArray(new ColumnStatisticsObj[0]);
+    for (int i = 0; i < columnStatisticsObj.length; i++) {
+      alterTableStmt.add(getAlterTableStmtPartitionColStat(columnStatisticsObj[i].getStatsData(),
+          columnStatisticsObj[i].getColName(),
+          tblName,
+          ptName,
+          dbName));
+      if (checkBitVectors(columnStatisticsObj[i].getStatsData())) {
+        ST command = new ST(EXIST_BIT_VECTORS_PARTITIONED);
+        command.add(DATABASE_NAME, dbName);
+        command.add(TABLE_NAME, tblName);
+        command.add(PARTITION_NAME, ptName);
+        command.add(COLUMN_NAME, columnStatisticsObj[i].getColName());
+        alterTableStmt.add(command.render());
+      }
+    }
+    return alterTableStmt;
+  }
+
+  public List<String> getDDLPlanForPartitionWithStats(Table table,
+      Map<String, List<Partition>> tableToPartitionList
+                                                     ) throws MetaException, HiveException {
+    List<String> alterTableStmt = new ArrayList<String>();
+    String tableName = table.getTableName();
+    for (Partition pt : tableToPartitionList.get(tableName)) {
+      alterTableStmt.add(getAlterTableAddPartition(pt));
+      alterTableStmt.add(getAlterTableStmtPartitionStatsBasic(pt));
+    }
+    String databaseName = table.getDbName();
+    List<String> partNames = new ArrayList<String>();
+    //TODO : Check if only Accessed Column Statistics Can be Retrieved From the HMS.
+    List<String> columnNames = getTableColumnNames(table);
+    tableToPartitionList.get(tableName).stream().forEach(p -> partNames.add(p.getName()));
+    Map<String, List<ColumnStatisticsObj>> partitionColStats =
+        Hive.get().getPartitionColumnStatistics(databaseName,
```

Comment:
These can be expensive calls to HMS. Add log statements? Or add PerfLogger around the methods invoking these from ExplainTask.